soc/arc/snps_nsim/CMakeLists.txt

```diff
@@ -6,48 +6,54 @@ if(COMPILER STREQUAL gcc)
   if(NOT CONFIG_64BIT)
     # ISA_ARCV2 & 32BIT
     zephyr_compile_options(-mno-sdata)
-    zephyr_compile_options_ifdef(CONFIG_CPU_ARCEM -mmpy-option=wlh1)
-    zephyr_compile_options_ifdef(CONFIG_CPU_ARCHS -mmpy-option=plus_qmacw)
     if(CONFIG_CPU_ARCHS)
       zephyr_compile_options_ifdef(CONFIG_FPU -mfpu=fpud_all)
+      zephyr_compile_options_ifdef(CONFIG_ARC_MPY_OPTION -mmpy-option=${CONFIG_ARC_MPY_OPTION})
     else()
       zephyr_compile_options_ifdef(CONFIG_FPU -mfpu=fpuda_all)
+      # when use GNU compiler, higher mpy-option is not supported
+      if(CONFIG_ARC_MPY_OPTION GREATER 6)
+        zephyr_compile_options(-mmpy-option=6)
+      else()
+      	zephyr_compile_options_ifdef(CONFIG_ARC_MPY_OPTION -mmpy-option=${CONFIG_ARC_MPY_OPTION})
+      endif()
```

Comment:
Even if we actually need this I'd prefer to have this logic in some Cmake function/macro in arc `.cmake` and call it from SoC `CMakeLists.txt`. We can call it `arc_finalize_mpy_compile_options` and put option for both GNU and MWDT toolchain in it.

Reply:
not sure - if we want a clean, generic solution, then I come back to my earlier suggestion: it would be better to have the option to use tcf files for all hardware specific compiler options/flags.